### PR TITLE
Feat/sidemenu nav

### LINKED ISSUE #455: Proposal for Forma36's website navigation
Hey!

While working on the documentation search last Friday, I played around the navigation to try to improve it a little bit.

The issues I was trying to solve is while scrolling down to read some documentation if you want to go to another page you would need to scroll back up to find the navigation again.
The same thing happens if you want to search for something or click on a social link (Blog, GitHub and Community).

I took inspiration in how other documentation websites deal with this, like:
- [CircuitUI](https://circuit.sumup.com/) from SumUp (Storybook Docs)
- [Atlassian design](https://atlassian.design/guidelines/product/foundations/typography)
- [Pajamas](https://design.gitlab.com/) from GitLab

And I propose moving all the navigation to the side menu. Keeping the logo, the social links and the search fixed in the top left. The list of links for the components documentation would scroll independently from the content (see the gif). Also, we can keep the three categories (Foundation, Guidelines and Components) always expanded.

### Home
Screenshot 2020-04-06 at 13 05 50

### Documentation Page
Screenshot 2020-04-06 at 13 06 09

### Scrolling
Apr-03-2020 19-25-51

### PR DESCRIPTION
**Pull request by gui-santos**
Screenshot 2020-04-16 at 15 12 39
Screenshot 2020-04-16 at 15 12 54

# Purpose of PR

This PR makes the side menu scrolling independent from the content scrolling. 
And it loads the three major categories already expanded 

Fixes #455

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/forma-36-website/src/components/Container.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ const styles = {
     width: 100%;
     display: flex;
     flex-direction: column;
+    height: calc(100vh - 56px);
```
[the remark this answers is not in the document]
Comment (gui-santos) [reply packages/forma-36-website/src/components/Container.js:30]: I totally agree! I think I did the same in another part but I missed this one hahaha

File: packages/forma-36-website/src/components/Navigation.js [hunk 2]
```diff
@@ -8,20 +8,30 @@ import { Icon } from '@contentful/forma-36-react-components';
 
 import DocSearch from './DocSearch';
 
+const heightOfHeader = 56;
```
[the remark this answers is not in the document]
Comment (gui-santos) [reply packages/forma-36-website/src/components/Navigation.js:11]: Ohhhh yes!

**Comment by gui-santos** [comment]
Thanks for reviewing this, @m10l 🙏 

> The smaller header size looks a bit too cramped now due to the size of the F36 logo and the text next to it. I'd think it's worth fixing or reverting it back to how it was.

Hmmm TBH, it was not my intention... I think I was brainstorming some stuff and forgot to revert the things the header. My bad! Thanks for spotting them

> The main navigation spacing looks a bit too extreme now. I'd revert back to using `spacingM`. There's also some spacing missing between the top of the sublist and the parent label.

I will revert the change and check the sublist spacing. I might have missed something while trying alternatives. Thanks

> Also, not sure if you've tested this already, but there's a Promo component which sits above the site, not sure if that has any impact. You can enable this by setting values for `promoText`, `promoLink`, and `promoLinkText` in `gatsby-config.js`

Ahhh I didn't know about this... I will make a test, and then I show you

**Comment by domarku** [comment]
Hi @gui-santos, thanks for the change, I love the idea to make the navigation scrollable.

I agree with @m10l about not expanding all the top-level items by default because it looks chaotic with the current styling. But (!) I like the idea because it presents a nice overview without having to click to expand (we might decide on a different approach as we grow the design system)
Why not then adjust the styling a bit? A couple of details I think we should consider:

- There should be a hover state on the menu items
- There shouldn't be such a strong difference between the items that have children
- There's not really a need for a background color in the navigation — if we set it to white it would improve the contrast

Additionally:
- The position of the search is not really correct, because it implies that the search is only going to search for items in the navigation, a sort of filter rather than a fully powered search across the whole website. I suggest we put the search in the top bar and trigger it with a button.

I made a quick sketch of the adjustments I have in mind — https://contentful.invisionapp.com/share/B9WW8W1NEUF

Thoughts @gui-santos @m10l ?

**Comment by m10l** [comment]
Nice @domarku, I like the adjustments in your sketch.

> There should be a hover state on the menu items

Agree, I think this should be distinct from the active style too.

> There shouldn't be such a strong difference between the items that have children

Totally agree. Makes sense for the top level headings, but not actual pages with children.

> There's not really a need for a background color in the navigation — if we set it to white it would improve the contrast

I think either use a border or a background colour to separate nav from the main content but not both. I quite like the background colour as it makes the main content easier to focus on

> The position of the search is not really correct, because it implies that the search is only going to search for items in the navigation, a sort of filter rather than a fully powered search across the whole website. I suggest we put the search in the top bar and trigger it with a button.


What you're saying makes sense, but I've seen the sidebar search placement as a common approach elsewhere. I don't think it's too confusing. I think @gui-santos originally started building the search similarly to how you suggested though.


In general though I think this is going too deep into another topic and needs more thought/discussion and we should just focus on the purpose of this PR - to make the side nav it's own scrollable container

**Comment by domarku** [comment]
> > There should be a hover state on the menu items
> Agree, I think this should be distinct from the active style too.
A difference in font-weight should do. Keep hover the same weight as idle, and make the active bold.

> > There's not really a need for a background color in the navigation — if we set it to white it would improve the contrast
> I think either use a border or a background colour to separate nav from the main content but not both. I quite like the background colour as it makes the main content easier to focus on
I don't have a strong opinion on this one.

> > The position of the search is not really correct, because it implies that the search is only going to search for items in the navigation, a sort of filter rather than a fully powered search across the whole website. I suggest we put the search in the top bar and trigger it with a button.
> What you're saying makes sense, but I've seen the sidebar search placement as a common approach elsewhere. I don't think it's too confusing. I think @gui-santos originally started building the search similarly to how you suggested though.
> In general though I think this is going too deep into another topic and needs more thought/discussion and we should just focus on the purpose of this PR - to make the side nav it's own scrollable container
I agree, and apologies for going too wide. I got into search placement because it was taking up significant vertical space at the top of the navigation for no particular reason, and then noticed that putting it above navigation doesn't really make a lot of sense because it searches the whole website.

-- 
If you want to proceed with adjusting the styles of the menu items a bit, here's the Invision page with the specs https://contentful.invisionapp.com/d/main/#/console/19784549/413884828/inspect

**Comment by gui-santos** [comment]
@m10l and @domarku 

>In general though I think this is going too deep into another topic and needs more thought/discussion and we should just focus on the purpose of this PR - to make the side nav it's own scrollable container

Yeaaaa... I think I should have left the expanded thing for another PR/Proposal. I agree it would work better in a different styling I made it this way as an intermediate stage

So...
I think @domarku could open a GitHub issue proposing those changes on the side menu and we work on them later. I really liked them! (I can open the issue if you are too busy 😄)

For now, I will revert the expanded thing and make the fixes I mentioned before

What do you guys think?

**Comment by m10l** [comment]
@gui-santos sounds good to me

**Pull request merged**